flow/errors.py

```diff
@@ -61,3 +61,9 @@ def parse(self, parser):
     def err(self, msg, caller):
         """Raise a template error."""
         raise jinja2.TemplateError(msg)
+
+
+class IncompatibleSchemaVersion(RuntimeError):
+    """The FlowProject's schema version is incompatible with this version of flow."""
```

Comment:
Pick one of the following. The second suggestion might make more sense, just depends on how we view the connection between the FlowProject and the filesystem. There is a clear definition of a signac project on disk and in memory (`Project`) but the association between a `FlowProject` and its configuration is a little less clear since `project.py` can be in another directory.
```suggestion
    """The FlowProject's schema version is incompatible with this version of signac-flow."""
```
```suggestion
    """The signac-flow configuration schema version is incompatible with this version of signac-flow."""
```

Reply:
I think that there are a couple of different questions we need to address here:

1. How is flow supposed to work if you put `project.py` in a different directory, unless you explicitly construct the `FlowProject` providing it the directory? You dug into this a bit in #196, but it looks like you never resolved the underlying issue since the problem somehow went away. 
2. Once we migrate to flow schema version 2, flow will have its own independent configuration. Should we require that configuration to live in the same directory as the `project.py` file? If not, how will we associate them?

One idea I've been toying with is whether config information should be on a per-FlowProject basis, rather than associated with a particular directory. That seems more consistent with some of the issues above regarding different directories, as well as the fact that multiple FlowProjects can be associated with a single signac project. This wouldn't resolve issues like where bundles are stored though, only the config variables.

I just opened glotzerlab/signac#541, which I think is also tangentially related to this discussion.

Reply:
I think enough of this is resolved in discussions on #506 that we can move forward with this PR now.